### DISCUSSION TITLE
Telescope Control: Protocol

### DISCUSSION THREAD

**Discussion by gtecac** [Stellarium/stellarium#643] [Q&A] [answered]
**Problem description:** 
Currently the telescope control protocol sends the following information (correct me if I am wrong):
1.  Length of packet (16bit)
2. Type(16 bit) (seems to be always zero) 
3. Time(64 bit)
4. RA(32 bit)
5. DEC(32 bit)

I wrote a small python server, running on a Raspberry Pi, which can receive these information and control a motor to follow a star which works perfectly. 
But when I want to track a satellite then these information are not enough, because the motor controller needs the satellite orbit. 

**Feature Request:** 
Expand the protocol by the "Two Line Elements (TLE)" information. 

**Help Request:** 
I already have downloaded the Stellarium Source Code and figured out that I only have to edit the `telescopeGoto()` function in the  TelescopeClient.cpp.  [TelescopeClient.cpp](https://github.com/Stellarium/stellarium/blob/c4c3552dbd67c6e50f4c035402eb3fc82fcd4354/plugins/TelescopeControl/src/TelescopeClient.cpp#L277) I don't know how to get the orbit(TLE) information from the function argument `selectObject ` ? Is there some kind of getOrbit() function ? 

`if (selectObject->getType() == "Satellite")
{
     selectObject->getOrbit() // Does not exist
}`

**Comment by alex-w** [comment, marked as answer]
See implementation of RTS2 client

**Comment by gtecac** [comment]
@alex-w : Tank you very much for your help ! That was exactly what I was looking for !
